File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnTrackEdit.java

package com.gracenote.gnsdk;

/** 
* 
*  represents  Track's information returned by Gracenote. 
*/ 
 
public class GnTrackEdit extends GnDataObject {
  private transient long swigCPtr;

  protected GnTrackEdit(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnTrackEdit_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnTrackEdit obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnTrackEdit(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnTrack gnTrack() {
    return new GnTrack(gnsdk_javaJNI.GnTrackEdit_gnTrack(swigCPtr, this), true);
  }

/** 
* 
*  Create an empty object of {@link GnTitleEdit} using TITLE OFFICIAL. 
*  @return {@link GnTitleEdit} 
*/ 
 
  public GnTitleEdit title() throws com.gracenote.gnsdk.GnException {
    return new GnTitleEdit(gnsdk_javaJNI.GnTrackEdit_title(swigCPtr, this), true);
  }

/** 
* 
*  Create an empty object of {@link GnArtistEdit} using ARTIST. 
*  @return {@link GnArtistEdit} 
*/ 
 
  public GnArtistEdit artist() throws com.gracenote.gnsdk.GnException {
    return new GnArtistEdit(gnsdk_javaJNI.GnTrackEdit_artist(swigCPtr, this), true);
  }

/** 
* 
*  Create an empty object of {@link GnCreditEdit} using CREDIT. 
*  @return {@link GnCreditEdit} 
*/ 
 
  public GnCreditEdit creditAdd() throws com.gracenote.gnsdk.GnException {
    return new GnCreditEdit(gnsdk_javaJNI.GnTrackEdit_creditAdd(swigCPtr, this), true);
  }

  public GnCreditEdit credit(long ord) throws com.gracenote.gnsdk.GnException {
    return new GnCreditEdit(gnsdk_javaJNI.GnTrackEdit_credit(swigCPtr, this, ord), true);
  }

/** 
* 
*  Create an empty object of {@link GnAudioWorkEdit} using AUDIO WORK. 
*  @return {@link GnAudioWorkEdit} 
*/ 
 
  public GnAudioWorkEdit audioWork() throws com.gracenote.gnsdk.GnException {
    return new GnAudioWorkEdit(gnsdk_javaJNI.GnTrackEdit_audioWork(swigCPtr, this), true);
  }

  public void mood(GnListElement moodElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnTrackEdit_mood(swigCPtr, this, GnListElement.getCPtr(moodElement), moodElement);
  }

  public void tempo(GnListElement tempoElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnTrackEdit_tempo(swigCPtr, this, GnListElement.getCPtr(tempoElement), tempoElement);
  }

  public void genre(GnListElement genreElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnTrackEdit_genre(swigCPtr, this, GnListElement.getCPtr(genreElement), genreElement);
  }

/** 
* 
*  Sets a list-based value by the list item Submit ID for YEAR. If the value does not 
*   exist, it is added. If the value does exist, it is changed. If the role_submit_id is 0, the 
*   value is deleted. 
*  @param value set Value corresponding to the specified {@link GnDataObject} value key 
*  <p><b>Remarks:</b></p> 
*  Use this function to set a list-based Submit ID to YEAR, prior to adding the {@link GnDataObject} to a 
*   parcel. 
*/ 
 
  public String year() {
    return gnsdk_javaJNI.GnTrackEdit_year__SWIG_0(swigCPtr, this);
  }

  public void year(String value) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnTrackEdit_year__SWIG_1(swigCPtr, this, value);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnTrackEdit.cs

namespace GracenoteSDK {

/**
*  @internal GnTrackEdit @endinternal
*  represents  Track's information returned by Gracenote.
*/
public class GnTrackEdit : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnTrackEdit(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnTrackEdit_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnTrackEdit obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnTrackEdit() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnTrackEdit(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnCreditEdit Credit(uint ord) {
    GnCreditEdit ret = new GnCreditEdit(gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Credit(swigCPtr, ord), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  public void Mood(GnListElement moodElement) {
    gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Mood(swigCPtr, GnListElement.getCPtr(moodElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Tempo(GnListElement tempoElement) {
    gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Tempo(swigCPtr, GnListElement.getCPtr(tempoElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Genre(GnListElement genreElement) {
    gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Genre(swigCPtr, GnListElement.getCPtr(genreElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  @internal Year @endinternal
*  Sets a list-based value by the list item Submit ID for YEAR. If the value does not
*   exist, it is added. If the value does exist, it is changed. If the role_submit_id is 0, the
*   value is deleted.
*  @param value set Value corresponding to the specified GnDataObject value key
*  <p><b>Remarks:</b></p>
*  Use this function to set a list-based Submit ID to YEAR, prior to adding the GnDataObject to a
*   parcel.
*/
  public string Year {
	/* csvarin typemap code */
	set 
	{
		System.IntPtr tempvalue = GnMarshalUTF8.NativeUtf8FromString(value);
		gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Year_set(swigCPtr, tempvalue);
		GnMarshalUTF8.ReleaseMarshaledUTF8String(tempvalue);
	}
 
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Year_get(swigCPtr) );
	} 

  }

/**
*  @internal getArtistEdit @endinternal
*  Create an empty object of GnArtistEdit using ARTIST.
*  @return GnArtistEdit
*/
  public GnArtistEdit Artist {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Artist_get(swigCPtr);
      GnArtistEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnArtistEdit(cPtr, true);
      return ret;
    } 
  }

/**
*  @internal CreditAdd @endinternal
*  Create an empty object of GnCreditEdit using CREDIT.
*  @return GnCreditEdit
*/
  public GnCreditEdit CreditAdd {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnTrackEdit_CreditAdd_get(swigCPtr);
      GnCreditEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnCreditEdit(cPtr, true);
      return ret;
    } 
  }

/**
*  @internal getWorkEdit @endinternal
*  Create an empty object of GnAudioWorkEdit using AUDIO WORK.
*  @return GnAudioWorkEdit
*/
  public GnAudioWorkEdit AudioWork {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnTrackEdit_AudioWork_get(swigCPtr);
      GnAudioWorkEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnAudioWorkEdit(cPtr, true);
      return ret;
    } 
  }

/**
*  @internal getTitleEdit @endinternal
*  Create an empty object of GnTitleEdit using TITLE OFFICIAL.
*  @return GnTitleEdit
*/
  public GnTitleEdit Title {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnTrackEdit_Title_get(swigCPtr);
      GnTitleEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnTitleEdit(cPtr, true);
      return ret;
    } 
  }

  public GnTrack GnTrack {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnTrackEdit_GnTrack_get(swigCPtr);
      GnTrack ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnTrack(cPtr, true);
      return ret;
    } 
  }

}

}
